client/src/templates/Challenges/classic/mobile-layout.tsx

```diff
@@ -45,6 +45,69 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
     });
   };
 
+  avoidWindowScrolling = (event: Event) => {
+    event.preventDefault();
+  };
+
+  setToolPanelPosition = () => {
+    const toolPanelGroup = (
+      document.getElementsByClassName(
+        'tool-panel-group-mobile'
+      ) as HTMLCollectionOf<HTMLElement>
+    )[0];
+
+    if (navigator.userAgent.match(/iPhone|Android.+Mobile/)) {
+      if (
+        visualViewport?.height &&
+        window.innerHeight > visualViewport.height
+      ) {
```

Comment:
Is this if statement for the keyboard and the other for the address bar. If so, it would be great to comment out your solution to make it more obvious.

Reply:
The former of the if statements is for avoiding a TypeScript error and the latter to detect the appearance of the virtual keyboard on iPhone, and I agree with you to comment out the solution so I will add.